WordPress/gutenberg
# Create block interactive template

## What?



Add an `interactive` template for the `@wordpress/create-block` package.

## Why?



So people can use this template to start testing the Interactivity API.

## How?



This PR is the consequence of these previous PR's: https://github.com/WordPress/gutenberg/pull/52508 and https://github.com/WordPress/gutenberg/pull/52549

**Some notes on this code:** 
- In order to be able to add a `view.js` from the template we need `create-block` to allow this file. That's why this PR also contains some changes to the `create-block` package
  - I think it makes sense that default blocks created with `create-block` package also has the `view.js` implemented as part of a basic scaffold
  - Tests of `create-block` package have been properly adapted to the addition of this new file
- As `create-block-interactive-template` is a new package a new `manifest.json` has been created so this new package is being taken into account for the [Block Editor Handbook](https://developer.wordpress.org/block-editor/)
- As `console.log` are not allowed with the default rules of the linter there's a directive added to packages/create-block/lib/templates/block/view.js.mustache to skip this rule and allow this command (when comitting the code)

## Testing Instructions






- Run the `@wordpress/create-block` package using:
  - `cd packages/create-block`
  - `node index.js --template ../create-block-interactive-template`
- Fill in the rest of the answers.
- Take a look at the files:
  - The `block.json` should have `supports.interactivity: true` and a `viewScript` field.
  - The `render.php` file should contain directives.
  - There should be a `view.js` file with an import of `@wordpress/interactivity` and a call to `store()`.

If you want to try that the interactivity works, you can use `wp-now`:

- `cd (installation directory) && npm start`
- `npx @wp-now/wp-now start` (in another terminal)
- Go to wp-admin
- Install Gutenberg 16.2 or higher
- Add a new post or edit an existing one
- Add the new block
- Save the post
- Go to the frontend
- Check that the toggle works
